src/ncdiff/netconf.py

```diff
@@ -983,18 +984,31 @@ def node_sub(self, node_self, node_other, depth=0):
                     e = etree.SubElement(
                         child_other, key, nsmap=key_node.nsmap)
                     e.text = key_node.text
+            if s_node.getparent().get('type') == 'case':
+                # key: choice node, value: case node
+                choice_nodes[s_node.getparent().getparent()] = s_node.getparent()
         for child_other in in_o_not_in_s:
             child_self = etree.Element(child_other.tag,
                                        {operation_tag: self.preferred_delete},
                                        nsmap=child_other.nsmap)
             if self.preferred_create != 'merge':
                 child_other.set(operation_tag, self.preferred_create)
             siblings = list(node_self.iterchildren(tag=child_other.tag))
+            s_node = self.device.get_schema_node(child_other)
             if siblings:
                 siblings[-1].addnext(child_self)
             else:
-                node_self.append(child_self)
-            s_node = self.device.get_schema_node(child_other)
+                # Append node if:
+                # Node not in case
+                # Node in case but choice node not in self
+                # Node in case and choice not in self but case also in self
+                if s_node.getparent().get('type') == 'case':
+                    choice_node = s_node.getparent().getparent()
+                    if choice_node not in choice_nodes or \
+                       s_node.getparent() == choice_nodes[choice_node]:
```

Comment:
`choice_node in choice_nodes and s_node.getparent() == choice_nodes[choice_node]:`

Reply:
Is "choice_node in choice_nodes and" necessary? If the logic reaches "s_node.getparent() == choice_nodes[choice_node]" we already know "choice_node in choice_nodes"

Reply:
You are right. No change is needed.